src/expression/visibility.ts

```diff
@@ -0,0 +1,43 @@
+import {createExpression, findGlobalStateRefs} from '.';
+import {type GlobalProperties, type StylePropertySpecification} from '..';
+import {type VisibilitySpecification} from '../types.g';
+
+const visibilitySpec: StylePropertySpecification = {
+    type: 'enum',
+    'property-type': 'data-constant',
+    expression: {
+        interpolated: false,
+        parameters: ['global-state']
+    },
+    values: {visible: {}, none: {}},
+    transition: false,
+    default: 'visible'
+};
+
+export default function createVisibility(visibility: VisibilitySpecification, globalState: Record<string, any>) {
+    const expression = {
+        setValue,
+        evaluate: null
+    };
+    setValue(visibility);
+    return expression;
+
+    function setValue(visibility: VisibilitySpecification) {
```

Comment:
Is there a reason this method is inside the other one?

Reply:
yes